src/debug.ts

```diff
@@ -157,6 +214,32 @@ export async function debugTests(
   await deferredPromise.promise
 }
 
+function getTestProjectsInFolder(path: string, api: VitestFolderAPI, tree: TestTree) {
+  const folder = tree.getOrCreateFolderTestItem(api, path)
+  const items = tree.getFolderFiles(folder)
+  return items.map(item => (getTestData(item) as TestFile).project)
+}
+
+function getProjectsFromTests(item: vscode.TestItem | undefined, api: VitestFolderAPI, tree: TestTree): string[] {
+  const items = getTestProjectsInFolder(item?.uri?.fsPath ?? '', api, tree)
+  if (items.length > 0) {
+    return items
+  }
+  // Climb up tree until entry with project is found
+  if (item?.parent) {
+    return getProjectsFromTests(item.parent, api, tree)
+  }
+  return []
+}
+
+function getBrowserModeLaunchArgs(isPlaywright: boolean, config: any): string {
+  const browser = isPlaywright ? 'chromium' : 'chrome'
+  // Only playwright provider supports --inspect-brk currently
+  const inspectBrk = isPlaywright ? `--inspect-brk=localhost:${config.debuggerPort ?? '9229'}` : ''
+  // regardless of user config, some properties need to be set when debugging with browser mode enabled
+  return `vitest ${config.cliArguments ?? ''} ${inspectBrk} --browser=${browser}`
```

Comment:
What happens if `cliArguments` already have these options? They will be converted into arrays here:

https://github.com/vitest-dev/vscode/blob/f0fa52bbb4acd425374b43972ebb8372431d8706/src/worker/init.ts#L21

Maybe we should normalize them afterwards just in case?